### ISSUE TITLE
Unexpected behaviour with the `weighted_sum` decision method [BUG] 

### ISSUE THREAD

**Issue by tsmbland** [EnergySystemsModellingLab/MUSE_OS#317]
## Describe the bug

I've been getting some unexpected behaviour using multiple objectives with the `weighted_sum` decision method, so I just thought I'd run a few sanity checks.

Firstly, for reference, here's how the default model looks with LCOE as a single objective:

![capacity_lcoe](https://github.com/EnergySystemsModellingLab/MUSE_OS/assets/23723407/b609fae9-3cf1-4559-8eca-205a8fa707db)

I would think that using the `weighted_sum` decision method and adding a second 'dummy' objective with a weight of zero shouldn't have any impact on the simulation, but it actually changes the results quite substantially:

![capacity_multi_objective_100_LCOE_0_fuel_consumption_cost](https://github.com/EnergySystemsModellingLab/MUSE_OS/assets/23723407/1d90918b-728d-4b86-b8e2-2bb2494d7931)

Similarly, having two LCOE objectives with 0.5 weight also has the same effect on the results:

![capacity_multi_objective_50_LCOE_50_LCOE](https://github.com/EnergySystemsModellingLab/MUSE_OS/assets/23723407/7e8a381b-a051-425b-80e9-87a1d75e8802)

Am I being overly simplistic to think that these two tests should be identical to the default example, or is there something weird going on here?

## Context

Please, complete the following to better understand the system you are using to run MUSE.

- Operative system (eg. Windows 10): windows 10
- MUSE version (eg. 1.0.1): dev
- Installation method (eg. pipx, pip, development mode): dev
- Python version (you can get this running `python --version`): 3.9

**Comment by tsmbland** [comment]
@dalonsoa

**Comment by dalonsoa** [comment]
This is indeed odd. I think it might be related to that normalisation [0, 1] that the `weighted_sum` method is doing. 

I would suggest your debug the problem starting by checking the `objectives` and `decision` objects in the following method in both scenarios:

https://github.com/EnergySystemsModellingLab/MUSE_OS/blob/7261daa746dd28e9814d9ea3c48e3cdc31530399/src/muse/agents/agent.py#L291

To start with, if you have two LCOE objectives, for example, the `objectives` objects should reflect two of the same thing, and both identical to the case of having just one LCOE. Assuming that is the case, then dig into the `weighted_sum` to understand how those two are transformed and why the result is different than a single LCOE. 

Unless I'm also being naive and `weighted_sum` is actually expected to do something different, I think this should point us to what is wrong here.

**Comment by ahawkes** [comment]
Yup, this sounds very odd, would appreciate some digging!
________________________________

**Comment by tsmbland** [comment]
I think the problem here is the normalization to [0, 1].

Let's say you have two objectives with equal weight that you're trying to minimise, and two technologies. Technology 1 has (10.1, 1) for the two objectives, whereas technology 2 has (10, 5).

Clearly technology 1 is the better choice (only slightly worse for objective 1 but much better for objective 2), however after normalization these become (1, 0) and (0, 1), so the technologies are seen as equally good (both 0.5 after the weighted average).

I get the need to normalise, as different metrics can have different scales, but there must be a better way of doing this.

Could simply divide by the maximum (absolute) value? With the above example you'd then have (1, 0.2) and (0.99, 1) for the two technologies, so clearly technology 1 would win. I think this would be sufficient since most metrics have a well-defined zero.

Or maybe some kind of log-normalization? But that wouldn't work with negative values

**Comment by tsmbland** [comment]
And just relating this to the initial observation with the dummy objectives, if you have the same objective twice, you might have for example (10.1, 10.1) and (10.0, 10.0), which becomes (1, 1), (0, 0). So, whereas in the single objective case technology 2 is only slightly better (10 vs 10.1), if using multiple objectives with normalization technology 2 is seen as much better (0 vs 1), so the results might be very different.

**Comment by ahawkes** [comment]
Good idea. May need to watch out for asymptotic instabilities (where the best objective is zero or close to zero).
